docgenerator/src/main/scala/codacy/codenarc/docs/ResourceHelper.scala

```diff
@@ -0,0 +1,51 @@
+package codacy.codenarc.docs
+
+import java.net.URL
+import java.nio.charset.{CodingErrorAction, StandardCharsets}
+import java.nio.file.{Files, Path, StandardOpenOption}
+
+import scala.io.{Codec, Source}
+import scala.util.{Failure, Properties, Try, Using}
+
+object ResourceHelper {
+
+  implicit val codec: Codec =
+    Codec.UTF8
+      .onMalformedInput(CodingErrorAction.REPLACE)
+      .onUnmappableCharacter(CodingErrorAction.REPLACE)
+
+  def getResourceContent(path: String): Try[List[String]] = {
+    Option(getClass.getClassLoader.getResource(path))
+      .map { url =>
+        getResourceContent(url)
+      }
+      .getOrElse {
+        Failure(new Exception("The path provided was not found"))
+      }
+  }
+
+  private def getResourceContent(url: URL): Try[List[String]] = {
+    Using.resource(url.openStream()) { stream =>
+      Try {
+        Source
+          .fromInputStream(stream)
+          .mkString
+          .split(Properties.lineSeparator)
+          .toList
+      }
+    }
```

Comment:
Here if you have an exception in `url.openStream()` it is thrown without getting catched from `Try`.
Probably it is not what you want.

Reply:
By the way you can use `Source.fromURL`

Reply:
Indeed it is not what I want. Changed the `Try`
Also, replaced the code to use `Source.fromURL`